src/serialization/container.h

```diff
@@ -38,19 +41,31 @@ namespace serialization
       return ::do_serialize(ar, e);
     }
 
-    template <typename Archive>
-    bool serialize_container_element(Archive& ar, uint32_t& e)
-    {
-      ar.serialize_varint(e);
-      return true;
+#define CONTAINER_VARINT_SERIALIZER(type) \
+    template <typename Archive> \
+    bool serialize_container_element(Archive& ar, type& e) \
+    { \
+      if (serialization_size_t_backward_compat && !typename Archive::is_saving() && !std::is_same<uint64_t, type>() && !std::is_same<uint32_t, type>()) \
```

Comment:
I think the backwards compatibility code here and elsewhere would benefit greatly from a comment explaining its rationale (maybe the `serialization_size_t_backward_compat` flag can have the comment attached and everything else can reference it).